pilot/cmd/pilot-agent/config.go

```diff
@@ -56,7 +58,17 @@ func constructProxyConfig() (meshconfig.ProxyConfig, error) {
 		proxyConfig = *meshConfig.DefaultConfig
 	}
 
-	proxyConfig.Concurrency = &types.Int32Value{Value: int32(concurrency)}
+	// If concurrency is unset, we will automatically set this based on CPU requests/limits for sidecars.
+	// For gateways, this will use all available CPUs.
+	// If explicitly set concurrency flag, this will be used.
+	if concurrency == 0 && role.Type == model.SidecarProxy {
+		byResources := determineConcurrencyOption()
```

Comment:
Could not see what you change here

Reply:
as mentioned in the PR description, this change should have no user facing impact for users, but will allow them to further customize the injection template.